### PR TITLE
Initial implementation of listTokenBalancesById.

### LINKED ISSUE #12: Rest endpoint - /api/v1/tokens/{tokenId}/balances
- Implement the endpoint with all filters
- Check the query time and add needed filters.
- Refer to https://mainnet-public.mirrornode.hedera.com/api/v1/docs/#/tokens/listTokenBalancesById

### LINKED ISSUE #13: Rest endpoint - /api/v1/tokens/{tokenId}/nfts
- Implement the endpoint with all filters
- Check the query time and add needed filters.
- Refer to https://mainnet-public.mirrornode.hedera.com/api/v1/docs/#/tokens/listNfts

### PR DESCRIPTION
**Pull request by MarcKriguerAtHedera**
The parts that are known not to work are indicated.

Signed-off-by: Marc Kriguer <marc.kriguer@swirldslabs.com>

### CODE REVIEW

File: src/main/java/com/swirlds/recordserver/TokensService.java [hunk 1]
```diff
@@ -239,4 +241,210 @@ private void getTokenById(ServerRequest request, ServerResponse response) {
...
+        final int limit = parseLimitQueryString(limitParam);
+        System.out.println("listTokenBalancesById: limit = " + limit);
+        final QueryParamUtil.WhereClause accountWhereClause = (accountIdQueryParam.isPresent() ?
+                QueryParamUtil.parseQueryString(QueryParamUtil.Type._long, "account_id", accountIdQueryParam.get()) :
+                null);
+        System.out.println("accountWhereClause = " + accountWhereClause);
+        // restrict query to only the tokenId passed in as a path parameter
+        final String tokenId = request.path().segments().get(0);
+        accountBalanceQueryParam.ifPresent(s -> whereClauses.add(
+                QueryParamUtil.parseQueryString(QueryParamUtil.Type._long, "balance", s)));
```
Comment (mgoelswirlds) [inline src/main/java/com/swirlds/recordserver/TokensService.java:281]: It's not just equals for balance, there can be other operators too. You might want to handle the other  >,< cases too.

File: src/main/java/com/swirlds/recordserver/TokensService.java [hunk 2]
```diff
@@ -239,4 +241,210 @@ private void getTokenById(ServerRequest request, ServerResponse response) {
...
+        final Optional<String> orderParam = request.queryParams().first("order");
+
+        // restrict query to only the tokenId passed in as a path parameter
+        final String tokenId = request.path().segments().get(0);
+
+        // build and execute query
+        final List<QueryParamUtil.WhereClause> whereClauses = new ArrayList<>();
+        whereClauses.add(QueryParamUtil.parseQueryString(QueryParamUtil.Type._long, "token_id",
+                request.path().segments().get(0)));
+        accountIdQueryParam.ifPresent(s ->
```
Comment (mgoelswirlds) [inline src/main/java/com/swirlds/recordserver/TokensService.java:396]: similarly here. account_id has other operators, not just equals.

**Review by mgoelswirlds — CHANGES_REQUESTED** [review]
I have added some comments. Please take a look. Let me know if you have any questions.

File: src/main/java/com/swirlds/recordserver/TokensService.java (hunk 1 above)
Comment (MarcKriguerAtHedera) [reply src/main/java/com/swirlds/recordserver/TokensService.java:281]: That line of code handles less than, less than or equals, greater than, and greater than or equals.  Not 100% sure about "not equals", but I know it handles the rest.

File: src/main/java/com/swirlds/recordserver/TokensService.java (hunk 2 above)
Comment (MarcKriguerAtHedera) [reply src/main/java/com/swirlds/recordserver/TokensService.java:396]: Same as above as well -- it handles =, <. <=, >, and >= for sure (and potentially != as well).

**Review by mgoelswirlds — APPROVED** [review]

**Pull request merged**